File: backend/PaulFunBlogger.Api/Entities/Media.cs
namespace PaulFunBlogger.Api.Entities;

public class Media
{
    public int Id { get; set; }
    public string FileName { get; set; } = string.Empty;
    public string FilePath { get; set; } = string.Empty;
    public long FileSize { get; set; }
    public string MimeType { get; set; } = string.Empty;
    public int UploadedBy { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User Uploader { get; set; } = null!;
}

File: backend-go/internal/models/media.go
package models

import "time"

type Media struct {
	ID         uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	FileName   string    `gorm:"not null;size:255" json:"fileName"`
	FilePath   string    `gorm:"not null;size:500" json:"filePath"`
	FileSize   int64     `gorm:"not null" json:"fileSize"`
	MimeType   string    `gorm:"not null;size:100" json:"mimeType"`
	UploadedBy uint      `gorm:"not null;index" json:"uploadedBy"`
	CreatedAt  time.Time `json:"createdAt"`

	// Association
	Uploader User `gorm:"foreignKey:UploadedBy" json:"uploader"`
}
